diff --git a/pdf.m b/pdf.m
index 3d620b5..35c7657 100644
--- a/pdf.m
+++ b/pdf.m
@@ -125,6 +125,7 @@ - (void)drawAt:(NSPoint)p
 	NSImage *out;
 	NSSize asize, bsize;
 	NSRect r;
+	NSImageInterpolation prevInterpolation;
 
 	asize = [artwork size];
 	bsize.width = width;
@@ -132,9 +133,13 @@ - (void)drawAt:(NSPoint)p
 	out = [[NSImage alloc] initWithSize:bsize];
 	[NSGraphicsContext saveGraphicsState];
 	[out lockFocus];
+	// this isn't saved by saveGraphicsState
+	prevInterpolation = [[NSGraphicsContext currentContext] imageInterpolation];
+	[[NSGraphicsContext currentContext] setImageInterpolation:NSImageInterpolationNone];
 	r.origin = NSZeroPoint;
 	r.size = bsize;
 	[artwork drawInRect:r];
+	[[NSGraphicsContext currentContext] setImageInterpolation:prevInterpolation];
 	[out unlockFocus];
 	[NSGraphicsContext restoreGraphicsState];
 	// TODO is this correct?
